pkgs/development/libraries/ffmpeg/generic.nix

```diff
@@ -92,6 +92,7 @@
 , withVmaf ? withFullDeps && !stdenv.isAarch64 && lib.versionAtLeast version "5" # Netflix's VMAF (Video Multi-Method Assessment Fusion)
 , withVoAmrwbenc ? withFullDeps && withVersion3 # AMR-WB encoder
 , withVorbis ? withHeadlessDeps # Vorbis de/encoding, native encoder exists
+, withVpl ? false # Hardware acceleration via intel libvpl
```

Comment:
How much extra closure size would this be if you enabled it now that the driver bit is impure?

Reply:
showing just the different lines:

```
# withMfx
/nix/store/556zw392nqnc0r8fg9n7yid1w920zd80-intel-media-sdk-23.2.2                 	  28.8M	  85.8M
/nix/store/mjndrx4agd3l69z4l0b98vv18dfsn9kb-ffmpeg-full-6.1-data                   	 264.9K	 264.9K
/nix/store/ndwfmpylljb6hjcpy6iz4557akc9ffmp-ffmpeg-full-6.1-lib                    	  27.7M	   1.3G
/nix/store/9ryljr6yh38sssfg4ajy3pv1c089xwzf-ffmpeg-full-6.1-bin                    	 727.2K	   1.3G

# withVpl
/nix/store/d8w5qfswmgxcjqwnmqw2v9r8amrvdlpb-xgcc-13.2.0-libgcc                     	 155.8K	 155.8K
/nix/store/vqvbn2z8wyrjwvayjb2vy5krhh1kis9b-libunistring-1.1                       	   1.7M	   1.7M
/nix/store/krqp9wj3rgalmqv04y0sqw987mxsnddn-libidn2-2.3.7                          	 352.7K	   2.1M
/nix/store/ksk3rnb0ljx8gngzk19jlmbjyvac4hw6-glibc-2.38-44                          	  28.9M	  31.1M
/nix/store/4vzal97iq3dmrgycj8r0gflrh51p8w1s-bash-5.2p26                            	   1.5M	  32.6M
/nix/store/alxp4kb0dgfczqwd7xfrpflhq50w0wck-gcc-13.2.0-libgcc                      	 155.8K	 155.8K
/nix/store/pp0jsd045xvfsz60kpbkfxbs9pbpk8z5-gcc-13.2.0-lib                         	   8.7M	  39.9M
/nix/store/b7gaf7alc4qjymv1mz79cshsayf1w20s-libvpl-2.10.1                          	   1.1M	  42.5M
/nix/store/vcyijj6xrw3acjphyb0sji1klz4xyslf-ffmpeg-full-6.1-data                   	 264.9K	 264.9K
/nix/store/smhwvfrq7j5khsk8761qiyryn5lvp7qw-ffmpeg-full-6.1-lib                    	  27.8M	   1.3G
/nix/store/rhmmr863zwlrxf8pvgfixml63zl87iwd-ffmpeg-full-6.1-bin                    	 727.2K	   1.3G
```

and when I change to `withVpl` for my full system closure it's a 26 M improvement:
```
<<< /run/current-system
>>> /nix/store/r0frim4jr544rkh0fxyk7rsbd1hmqr5z-nixos-system-coffee-23.11.20240308.2be119a
Version changes:
[C.]  #1  bash  5.2-p15 x2, 5.2-p21 x2, 5.2p26 -> 5.2-p15 x2, 5.2-p21 x2, 5.2p26 x2
Added packages:
[A.]  #1  libvpl  2.10.1
Removed packages:
[R.]  #1  intel-media-sdk  23.2.2
Closure size: 2763 -> 2764 (24 paths added, 23 paths removed, delta +1, disk usage -26.2MiB).
```

Reply:
Neat! Once GPUs only supported by MFX are sufficiently legacy (or if ffmpeg were to support both), I see nothing blocking its enablement then.

Reply:
thanks for all your help getting this PR merged! 😄